src/Umbraco.PublishedCache.HybridCache/SeedKeyProviders/Document/DocumentBreadthFirstKeyProvider.cs

```diff
@@ -6,10 +6,74 @@ namespace Umbraco.Cms.Infrastructure.HybridCache.SeedKeyProviders.Document;
 
 internal sealed class DocumentBreadthFirstKeyProvider : BreadthFirstKeyProvider, IDocumentSeedKeyProvider
 {
+    private readonly IDocumentNavigationQueryService _navigationQueryService;
+    private readonly IPublishStatusQueryService _publishStatusService;
+    private readonly int _seedCount;
+
     public DocumentBreadthFirstKeyProvider(
         IDocumentNavigationQueryService documentNavigationQueryService,
-        IOptions<CacheSettings> cacheSettings)
+        IOptions<CacheSettings> cacheSettings,
+        IPublishStatusQueryService publishStatusService)
         : base(documentNavigationQueryService, cacheSettings.Value.DocumentBreadthFirstSeedCount)
     {
+        _navigationQueryService = documentNavigationQueryService;
+        _publishStatusService = publishStatusService;
+        _seedCount = cacheSettings.Value.DocumentBreadthFirstSeedCount;
+    }
+
+    public new ISet<Guid> GetSeedKeys()
```

Comment:
Maybe better if this was virtual on the base class so could be overridden?  But I guess you can't change that as the base class is public.  Perhaps a `TODO` for V16 would be good though?

Reply:
Would be even better if there was a way to reuse the functionality of the base class rather than having to copy it here.    Perhaps you could have a virtual method on the base class called `FilterRootKeys`, that in the base implementation does nothing, but you can override it?  Then you'd only need one line of code in this derived class.  Again though, that might be tricky to do in a non-breaking way.

Reply:
Oh that's a great idea with the filtering virtual method, but yes, I do believe this is a breaking change to introduce that virtual method, it's unlikely to cause issues, but I think it's technically breaking, so I've added a TODO for v16, along with a comment on how this could be done, I should probably just do it after this has been merged though :smile: